### PR TITLE
Refactor timer repro part2

### PR DESCRIPTION
**Pull request by tevoinea**

### CODE REVIEW

File: src/ApiService/ApiService/OneFuzzTypes/Enums.cs [hunk 1]
```diff
@@ -192,7 +192,7 @@ public static class VmStateHelper {
             return
                 new[]{
                 VmState.Init,
-                VmState.ExtensionsLaunched,
+                VmState.ExtensionsLaunch,
```
Comment (tevoinea) [inline src/ApiService/ApiService/OneFuzzTypes/Enums.cs:195]: This was a typo

File: src/ApiService/ApiService/onefuzzlib/NsgOperations.cs [hunk 2]
```diff
@@ -3,24 +3,24 @@
...
         bool OkToDelete(HashSet<string> active_regions, string nsg_region, string nsg_name);
         Async.Task<bool> StartDeleteNsg(string name);
 
-        Async.Task DissociateNic(NetworkInterfaceResource nic);
+        Async.Task<ResultVoid<Error>> DissociateNic(Nsg nsg, NetworkInterfaceResource nic);
     }
 
 
-    public class Nsg : INsg {
+    public class NsgOperations : INsgOperations {
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/NsgOperations.cs:17]: Renamed `Nsg` -> `NsgOperations` to make room to create a record named `Nsg` to follow our convention.

File: src/ApiService/ApiService/onefuzzlib/DiskOperations.cs [hunk 3]
```diff
@@ -19,8 +20,18 @@ public class DiskOperations : IDiskOperations {
...
+    public async Task<bool> DeleteDisk(string resourceGroup, string name) {
+        try {
+            _logTracer.Info($"deleting disks {resourceGroup} : {name}");
+            var disk = await _creds.GetResourceGroupResource().GetDiskAsync(name);
+            if (disk != null) {
+                await disk.Value.DeleteAsync(WaitUntil.Started);
+                return true;
+            }
+        } catch (Exception e) {
+            _logTracer.Error($"unable to delete disk: {name} {e.Message}");
```
Comment (stishkin) [inline src/ApiService/ApiService/onefuzzlib/DiskOperations.cs:32]: there is also _logTracer.Exception(e)
(better to do both Error and Exception (or maybe modify log tracer to do both when Exception is logged...)

File: src/ApiService/ApiService/onefuzzlib/NsgOperations.cs [hunk 4]
```diff
@@ -46,8 +46,54 @@ public class Nsg : INsg {
...
-        public System.Threading.Tasks.Task DissociateNic(NetworkInterfaceResource nic) {
-            throw new NotImplementedException();
+        public async Async.Task<ResultVoid<Error>> DissociateNic(Nsg nsg, NetworkInterfaceResource nic) {
+            if (nic.Data.NetworkSecurityGroup == null) {
+                return ResultVoid<Error>.Ok();
+            }
+
+            var azureNsg = await GetNsg(nsg.Name);
+            if (azureNsg == null) {
+                return new ResultVoid<Error>(new Error(
```
Comment (stishkin) [inline src/ApiService/ApiService/onefuzzlib/NsgOperations.cs:56]: for this there is OneFuzzResult <- it populates error stuff for you

**Review by stishkin — APPROVED** [review]

**Review by chkeita — APPROVED** [review]

**Review by raghavthind2005 — APPROVED** [review]

**Pull request merged**